Flaky integration test: TestScenario_SexModeActivationDeactivationCycle

## Problem

`TestScenario_SexModeActivationDeactivationCycle` intermittently fails in the integration test suite. The failure is timing-dependent — it passes when run individually but fails when run alongside other integration tests.

**Failure output:**
```
--- FAIL: TestScenario_SexModeActivationDeactivationCycle (4.49s)
    scenario_sexmode_test.go:575:
        Error: Condition never satisfied
        Messages: Music should be restored to 'evening'
    scenario_sexmode_test.go:579:
        Error: Condition never satisfied
        Messages: Evening scene should be activated
    scenario_sexmode_test.go:581:
        Error: Expected value not to be nil.
        Messages: Evening scene should be activated
```

The test expects music and lighting to be restored after sex mode deactivation, but the state restoration doesn't complete within the wait timeout under load.

## Impact

- Blocks `git push` via pre-push hook when the flaky test happens to fail
- Forces use of `--no-verify` to push unrelated changes (e.g. #709)
- Erodes confidence in the test suite

## Suggested investigation

1. **Root cause this test**: Determine if the timeout in `helpers_test.go:53` / `helpers_test.go:81` is too tight, or if there's a genuine race condition in sexmode deactivation state restoration
2. **Audit other integration tests for similar patterns**: Any test using `WaitForCondition` or similar polling helpers with tight timeouts may be similarly flaky under load. Candidates to check:
   - Other scenario tests with state restoration assertions
   - Tests that depend on multiple plugins reacting in sequence
   - Tests with sub-second or low-second timeouts for cross-plugin state propagation
3. **Consider increasing timeouts or adding retry logic** for integration tests that assert on eventually-consistent state

## Reproduction

The failure is intermittent. Running the full integration suite several times will reproduce it:
```bash
make cache-clear-integration && make integration-tests
```

Running the test in isolation reliably passes:
```bash
cd homeautomation-go && go test -race -v -run TestScenario_SexModeActivationDeactivationCycle ./test/integration/...
```

--- a/homeautomation-go/test/integration/helpers_test.go
+++ b/homeautomation-go/test/integration/helpers_test.go
@@ -1,6 +1,7 @@
 package integration
 
 import (
+	"os"
 	"sync/atomic"
 	"testing"
 	"time"
@@ -16,15 +17,28 @@ import (
 // ============================================================================
 
 const (
-	// stateWaitTimeout is the maximum time to wait for state changes.
-	// This should be long enough to handle slow CI environments but short
-	// enough that tests don't hang for too long on actual failures.
-	stateWaitTimeout = 2 * time.Second
+	// defaultStateWaitTimeout is the default maximum time to wait for state changes.
+	// This should be long enough to handle slow CI environments (where many parallel
+	// tests compete for CPU) but short enough that tests don't hang too long on
+	// actual failures. Override with TEST_WAIT_TIMEOUT env var (e.g. "10s").
+	defaultStateWaitTimeout = 5 * time.Second
 
 	// statePollInterval is how often to check if the expected state is reached.
 	// Smaller values make tests faster in normal conditions but increase CPU usage.
 	statePollInterval = 10 * time.Millisecond
 )
+
+// stateWaitTimeout is the maximum time to wait for state changes.
+// Initialized from TEST_WAIT_TIMEOUT env var or defaultStateWaitTimeout.
+var stateWaitTimeout = defaultStateWaitTimeout
+
+func init() {
+	if v := os.Getenv("TEST_WAIT_TIMEOUT"); v != "" {
+		if d, err := time.ParseDuration(v); err == nil {
+			stateWaitTimeout = d
+		}
+	}
+}
 
 // ============================================================================
 // State Polling Helpers

--- a/homeautomation-go/test/integration/scenario_sleephygiene_test.go
+++ b/homeautomation-go/test/integration/scenario_sleephygiene_test.go
@@ -517,7 +517,7 @@ func TestScenario_WakeSequence_VolumeFadesOutMonotonically(t *testing.T) {
 	select {
 	case <-done:
 		t.Log("Fade-out completed")
-	case <-time.After(2 * time.Second):
+	case <-time.After(stateWaitTimeout):
 		t.Fatal("Fade-out did not complete within timeout")
 	}
 

